### PR TITLE
Add electricity-free soldering

### PR DESCRIPTION
**Pull request by AtomicFox556**
#### Summary
Content "Add electricity-free soldering"

#### Purpose of change
Add a simple-to-make tool to solder electronics without having a soldering iron, which may not be easy to find/make.

#### Describe the solution
Add a makeshift soldering iron which is basically a wire or nail secured to a piece of wood that would be heated over a heat source to do the soldering.
Removed soldering tool as a possible alternative to welder for recipes in which it'd not make sense.

#### Describe alternatives you've considered
None.

#### Testing
Launched the game, made sure that no errors happen.

#### Additional context
Based on https://www.youtube.com/watch?v=WR5L_C0EPyM

### CODE REVIEW

**Comment by AtomicFox556** [comment]
No idea why the tests are failing, everything works well in game. Can anyone help?

**Comment by Qrox** [comment]
There has been some previous discussion on it: #48391. It appears you would need much more material at the tip to retain enough heat for soldering.

File: data/json/recipes/armor/head.json [hunk 1]
```diff
@@ -1608,7 +1608,7 @@
     "autolearn": true,
     "using": [ [ "adhesive", 1 ] ],
     "qualities": [ { "id": "CUT", "level": 2 }, { "id": "FABRIC_CUT", "level": 2 } ],
-    "tools": [ [ [ "welder", 56 ], [ "welder_crude", 84 ], [ "soldering_iron", 84 ], [ "toolset", 84 ] ] ],
```
Comment (Qrox) [inline data/json/recipes/armor/head.json:1611]: Why is soldering iron removed in these recipes?

File: data/json/recipes/recipe_others.json [hunk 2]
```diff
@@ -967,7 +967,8 @@
       { "id": "SCREW", "level": 1 },
       { "id": "WRENCH", "level": 1 }
     ],
-    "tools": [ [ [ "welder", 200 ], [ "welder_crude", 300 ], [ "toolset", 300 ], [ "soldering_iron", 300 ], [ "oxy_torch", 40 ] ] ],
+    "using": [ [ "soldering_standard", 10 ] ],
```
Comment (Qrox) [inline data/json/recipes/recipe_others.json:970]: Why is this added here, but not in other recipes?

File: data/json/requirements/toolsets.json [hunk 3]
```diff
@@ -120,7 +120,8 @@
     "id": "soldering_standard",
     "type": "requirement",
     "//": "Soldering metal items",
-    "tools": [ [ [ "soldering_iron", 1 ], [ "toolset", 1 ] ] ],
+    "qualities": [ { "id": "SOLDER", "level": 1 } ],
+    "tools": [ [ [ "soldering_iron", 1 ], [ "toolset", 1 ], [ "surface_heat", 2, "LIST" ], [ "lighter", 2 ], [ "ref_lighter", 2 ] ] ],
```
Comment (Qrox) [inline data/json/requirements/toolsets.json:124]: Does this actually require you to use either a proper soldering iron or a improvised one by having a fire source? Because I don't see `soldering_iron_improvised` referenced anywhere.

File: data/json/recipes/recipe_others.json (hunk 2 above)
Comment (AtomicFox556) [reply data/json/recipes/recipe_others.json:970]: This recipe has electronic parts in it, other similar recipes that were changed don't.

**Comment by AtomicFox556** [comment]
> There has been some previous discussion on it: #48391. It appears you would need much more material at the tip to retain enough heat for soldering.

As the video shows, the idea is that you'd need to heat up your tool after every single (or two, at most) connections soldered. You do need some more material if you're using a large fire, but this can be achieved by folding the piece of wire several times.

**Comment by Shodan14** [comment]
Solder quality 1 and quality 2 maybe?

**Comment by kasanryukin** [comment]
As I said in #48391, there are already non-electric options that exist that aren't as crude as you want to make one.   In terms of your design, it functionally won't work; it won't retain enough heat to actually solder anything more than two narrow gauge wires together.  The iron mentioned in #48391 was designed to retain a ton of heat to melt very low temp solder for things like whitesmithing, pipe fitting, and goldsmithing.

The logical option for an electricity free iron usable with-in the context of what the game uses soldering irons for, is a butane soldering iron.

**Comment by AtomicFox556** [comment]
> As I said in #48391, there are already non-electric options that exist that aren't as crude as you want to make one. In terms of your design, it functionally won't work; it won't retain enough heat to actually solder anything more than two narrow gauge wires together. The iron mentioned in #48391 was designed to retain a ton of heat to melt very low temp solder for things like whitesmithing, pipe fitting, and goldsmithing.
> 
> The logical option for an electricity free iron usable with-in the context of what the game uses soldering irons for, is a butane soldering iron.

Recipe updated as per the issue. I can't do anything regarding the fact that a proper soldering iron would also be usable as a soldering tool if you have a source of heat; if it's not acceptable like that, I'll scrap this and implement a butane-powered soldering iron later.

**Comment by AtomicFox556** [comment]
I think I'll try making a lamp-oil-heated soldering iron later instead. Would be awkward to implement all the possible heating options this way, but it would let wilderness-only characters do soldering if I can also figure out a way to produce basic electric components from raw materials.
